### PR TITLE
Bugfix/update owner ref behavior 4.0

### PR DESCRIPTION
**Pull request by shawn-hurley**
**Description of the change:**

* Ansible operator would create resources with owner references that were incorrect.
* This uses a new mechanism to dynamically watch resources that cross the namespace boundaries.

**Motivation for the change:**

* This was a very bad assumption on our part and is causing issues.

### CODE REVIEW

**Comment by jmrodri** [comment]
Typo in first commit message: accross
Typo in PR title: bahavior

**Comment by shawn-hurley** [comment]
@jmrodri Can you take another look

File: pkg/ansible/proxy/proxy.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ import (
 	"github.com/operator-framework/operator-sdk/pkg/ansible/proxy/controllermap"
 	"github.com/operator-framework/operator-sdk/pkg/ansible/proxy/kubeconfig"
 	k8sRequest "github.com/operator-framework/operator-sdk/pkg/ansible/proxy/requestfactory"
+	osdkHandler "github.com/operator-framework/operator-sdk/pkg/handler"
```
Comment (jmrodri) [inline pkg/ansible/proxy/proxy.go]: osdkHandler looks ugly but I'm sure gofmt will enforce it. I would prefer osdkhandler but won't hold this up for that.

File: pkg/ansible/proxy/proxy.go [hunk 2]
```diff
@@ -152,25 +152,8 @@ func CacheResponseHandler(h http.Handler, informerCache cache.Cache, restMapper
...
-							if oRef.APIVersion == ownerRef.APIVersion && oRef.Kind == ownerRef.Kind {
-								err := addWatchToController(ownerRef, cMap, un, restMapper)
-								if err != nil {
-									log.Error(err, "Could not recover dependent resource watch", "owner", ownerRef)
-									return
-								}
-							}
-						}
-					}()
+					go recoverDependentWatches(req, un, cMap, restMapper)
```
Comment (jmrodri) [inline pkg/ansible/proxy/proxy.go]: :+1:

File: pkg/ansible/proxy/proxy.go [hunk 3]
```diff
@@ -248,7 +263,33 @@ func InjectOwnerReferenceHandler(h http.Handler, cMap *controllermap.ControllerM
 				http.Error(w, m, http.StatusBadRequest)
 				return
 			}
-			data.SetOwnerReferences(append(data.GetOwnerReferences(), owner.OwnerReference))
+
+			addOwnerRef, err := shouldAddOwnerRef(data, owner, restMapper)
+			if err != nil {
+				m := "Could not determine if we should add owner ref"
+				log.Error(err, m)
+				http.Error(w, m, http.StatusBadRequest)
```
Comment (jmrodri) [inline pkg/ansible/proxy/proxy.go]: If this http error is expected then okay.

**Pull request merged**